thrift.js

```diff
@@ -46,17 +48,32 @@ var ThriftMap = require('./map').ThriftMap;
 var ThriftConst = require('./const').ThriftConst;
 var ThriftTypedef = require('./typedef').ThriftTypedef;
 
+var validThriftIdentifierRE = /^[a-zA-Z0-9_][a-zA-Z0-9_\.]+$/;
+
 function Thrift(options) {
     var self = this;
 
     assert(options, 'options required');
     assert(typeof options === 'object', 'options must be object');
-    assert(options.source, 'source required');
-    assert(typeof options.source === 'string', 'source must be string');
+    assert(options.source || options.thriftFile,
+        'opts.source or opts.thriftFile required');
+
+    self.thriftFile = options.thriftFile ?
+        path.resolve(options.thriftFile) : null;
+
+    if (options.source) {
+        assert(typeof options.source === 'string', 'source must be string');
+        self.source = options.source;
+    }
+
+    if (self.thriftFile && !self.source) {
+        self.source = fs.readFileSync(self.thriftFile, 'ascii');
+    }
 
     self.strict = options.strict !== undefined ? options.strict : true;
 
     self.claims = Object.create(null);
+    self.definitions = Object.create(null);
```

Comment:
I'm down with merging claims and definitions. I went with definitions since that is what the thrift IDL calls these declarations.